tensorly/tests/test_parafac2_tensor.py

```diff
@@ -0,0 +1,203 @@
+import numpy as np
+import pytest
+
+from .. import backend as tl
+from ..base import unfold, tensor_to_vec
+from ..parafac2_tensor import (parafac2_to_tensor, parafac2_to_unfolded,
+                               parafac2_to_vec, _validate_parafac2_tensor,
+                               parafac2_to_slice, parafac2_to_slices,
+                               parafac2_normalise, apply_parafac2_projections)
+from ..tenalg import kronecker, mode_dot
+from ..testing import (assert_array_equal, assert_array_almost_equal, 
+                       assert_equal, assert_raises)
+from ..random import check_random_state, random_parafac2
+
+
+def test_validate_parafac2_tensor():
+    rng = check_random_state(12345)
+    true_shape = [(4, 5)]*3
+    true_rank = 2
+    weights, factors, projections = random_parafac2(true_shape, rank=true_rank)
+    
+    # Check shape and rank returned
+    shape, rank = _validate_parafac2_tensor((weights, factors, projections))
+    assert_equal(shape, true_shape,
+                 err_msg='Returned incorrect shape (got {}, expected {})'.format(
+                    shape, true_shape))
+    assert_equal(rank, true_rank,
+                 err_msg='Returned incorrect rank (got {}, expected {})'.format(
+                    rank, true_rank))
+
+    # One of the factors has the wrong rank
+    for mode in range(3):
+        false_shape = (len(factors[mode]), true_rank+1)
```

Comment:
Another TensorFlow issue ... 
```suggestion
        false_shape = (tl.shape(factors[mode])[0], true_rank+1)
```